crates/tasks/src/lib.rs

```diff
@@ -251,15 +251,24 @@ impl TaskExecutor {
     }
 
     /// Spawns a future on the tokio runtime depending on the [TaskKind]
-    fn spawn_on_rt<F>(&self, fut: F, task_kind: TaskKind) -> JoinHandle<()>
+    fn spawn_on_rt<F>(&self, fut: F, task_kind: TaskKind, name: &'static str) -> JoinHandle<()>
```

Comment:
Please make this an `Option` instead to make the behavior of no name more explicit